validator/client/attest_protect.go

```diff
@@ -26,46 +27,41 @@ func (v *validator) slashableAttestationCheck(
 	ctx, span := trace.StartSpan(ctx, "validator.postAttSignUpdate")
 	defer span.End()
 
+	// Based on EIP3076, validator should refuse to sign any attestation with source epoch less
+	// than the minimum source epoch present in that signer’s attestations.
+	lowestSourceEpoch, exists, err := v.db.LowestSignedSourceEpoch(ctx, pubKey)
+	if err != nil {
+		return err
+	}
+	if exists && lowestSourceEpoch > indexedAtt.Data.Source.Epoch {
+		return fmt.Errorf(
+			"could not sign attestation lower than lowest source epoch in db, %d > %d",
+			lowestSourceEpoch,
+			indexedAtt.Data.Source.Epoch,
+		)
+	}
+	existingSigningRoot, err := v.db.SigningRootAtTargetEpoch(ctx, pubKey, indexedAtt.Data.Target.Epoch)
+	if err != nil {
+		return err
+	}
+	signingRootsDiffer := slashutil.SigningRootsDiffer(existingSigningRoot, signingRoot)
```

Comment:
Should you always return early if this is true? Similar to @rkapka's comment.

Reply:
No because the slashing condition only applies if we have the same target epoch but different signing root, so we need to perform the check below as well